tox.ini

```diff
@@ -0,0 +1,17 @@
+[tox]
+envlist = py27,py36,py37,py38
+
+[testenv]
+sitepackages=False
+deps = pytest               # PYPI package providing pytest
+commands = python -m py.test tests
+passenv = *
+extras =
+    testing
+
+[testenv:py37]
+setenv =
+    PYTHONDONTWRITEBYTECODE=1
```

Comment:
Why?

Reply:
sorry, that's try and error, till I moved tests outside package, seems like pytest & tox doesn't like it. And found this at stackoverflow, that obviously don't worked.

Reply:
Whoa, I didn't notice that you moved the tests outside the package.

I don't like that; other zopefoundation packages conventionally use zope.testrunner and embed the tests inside the 'tests' subpackage of every package.  I thought z3c.alchemy was special.  I didn't want to insist on bringing it inline with the rest of ZF, but I am against bringing it out of line.

Reply:
If you need a working example, we have plenty of ZF packages using zope.testrunner from tox.  Here's how [zope.sqlalchemy does that](https://github.com/zopefoundation/zope.sqlalchemy/blob/053587de5c811aeefcba890148dbcb1aa20f1950/tox.ini).

Reply:
@jordic in general could you please respect conventions you found in this package instead of making large changes to code style, package structure etc? Only change what's needed.

Reply:
Sorry, I agree with both of you, I acted like if I were the owner of the package and applied what I'm used to do (without too much thinking that this is a package that's inside an organization, and the best is to follow the guidelans of it). 

On another side, I will pretty appreciate if you can point me on how to install a package like this using buildout, and being able to run test using zope.testrunner. I just arrived to the zope world too late, but I will apreciate a lot your effort helping me learn all of this tools and conventions. So, many thanks for your time reviewing and taking care of this :)

Reply:
Found it, the key is to just launch `python bootstrap.py`. 
```
https://github.com/zopefoundation/BTrees/blob/master/bootstrap.py#L83-L86
```

Reply:
This line on the original doesn't work. The script at http://peak.telecommunity.com/dist/ez_setup.py. Doesn't seem to be py3 friendly. Any hints on fixing this? 
https://github.com/zopefoundation/z3c.sqlalchemy/blob/master/bootstrap.py#L28-L30

Reply:
I wouldn't worry about buildout too much, myself.  I've been adding tox.ini's to any ZF package I work on, because otherwise running tests is too painful.

python bootstrap.py has bitrotted quite badly in recent times and people are now recommending to install zc.buildout in a virtualenv somewhere and to run `buildout bootstrap` instead.  People are also removing bootstrap.py files from ZF repos.  buildout.cfg is usually left alone, let people who like buildout use it if they want to.

Reply:
@mgedmin I totally disagree. buildout is still the standard.

bootstrap.py will only work under Python 2 and its use is deprecated. To work on a package and test we now use virtual environments and install zc.buildout into it before running the buildout, something like this:

```
git clone mypackage
cd mypackage
python3 -m venv .
bin/pip install -U pip wheel zc.buildout
bin/buildout
bin/test
```